### PR TITLE
Use pytest to run Python bindings tests

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #69 

This PR solves the issue by installing pytest and running the tests with it. Low risk change here that simply enables easier tests for the future

## Review guidelines

Estimated Time of Review: 2 minutes

### CODE REVIEW

File: python/Dockerfile.python [hunk 1]
```diff
@@ -23,6 +25,6 @@ RUN pip3 install .
 FROM build as testing
 
 COPY test test
-RUN python3 -m unittest
+RUN pytest
```
Comment (eeberhard) [inline python/Dockerfile.python:28]: Would something like this work for the interim?
```suggestion
# TODO: remove unittest after refactor to purest
RUN python3 -m unittest
RUN pytest
```

**Review by eeberhard — COMMENTED** [review]
I imagine you won't be able to refactor all the tests in one go, so that there will be some `unittest` and some `pytest` cases. I think with dynamic test discovery you can probably run `unittest` and `pytest` back to back.

And, the `build-test-python` action still uses unittest, so when you start refactoring to use `pytest` you won't see the results in the CI. I think this would be a good place to also invoke `pytest` in that CI action. At the end of the test refactor you can then remove the `unittest` calls entirely.

Edit: as mentioned in the original issue #69, `pytest` will already discover and run `unittest` tests, so it's not necessary to invoke the test runner twice. Still, the CI should be updated.

**Comment by domire8** [comment]
> Edit: as mentioned in the original issue #69, `pytest` will already discover and run `unittest` tests, so it's not necessary to invoke the test runner twice. Still, the CI should be updated.

Thanks for thinking about the CI.
I see several possibilities:
1. Installing pytest in the build-test-python action Dockerfile and wait until the next release where pytest will be installed in the base Docker image
2. Dispatch the build-push workflow of the base Dockerfiles from this feature branch such that pytest is installed and then we don't need to install it in the action and the Python Dockerfile

**Comment by eeberhard** [comment]
> 2. Dispatch the build-push workflow of the base Dockerfiles from this feature branch such that pytest is installed and then we don't need to install it in the action and the Python Dockerfile

I think, given the very minor change to the base Dockerfile, I agree with manually building it from this branch. That way we don't have to remember to remove that line in a future step. If you do that, I would wait to close this PR until the image is built and the CI passes again.

**Review by eeberhard — APPROVED** [review]

**Pull request merged**